File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> indexArray = new HashMap<>();
        for(int i = 0; i < nums.length; ++i){
            indexArray.put(nums[i], i);
        }

        int diff = 0;
        for(int i = 0; i < nums.length; ++i){

            diff = target - nums[i];
            if(indexArray.containsKey(diff) && indexArray.get(diff) != i)
                return new int[] {i, indexArray.get(diff)};         
        }
        return new int[] {};
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        s = 0
        l = len(numbers) - 1
        sum = 0

        while s < l:
            sum = numbers[s] + numbers[l]
            print(sum)
            if sum == target:
                return [s+1,l+1]
            elif sum > target:
                l -= 1    
            else:
                s += 1
        return [-1,-1]   